### PR TITLE
set background jobs via sidekiq-scheduler

### PR DESCRIPTION
**Pull request by moizansari22**
worked on background jobs . added functionality of gem 'sidekiq-scheduler' and created and exectuted jobs as per project requirments.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -1,38 +1,14 @@
 class EventsController < ApplicationController
-  
   def index
     @events = Event.all
```
Comment (MuhammadRaza29) [inline app/controllers/events_controller.rb]:
```suggestion
    if param[:web_source].present?
        @events = Event.where("websource LIKE ?", params[:websource]
    else
      @events = Event.all
```

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -1,38 +1,14 @@
...
-    total_pages = 20
-    while i < total_pages
-      url = "https://www.visitberlin.de/en/event-calendar-berlin?page=#{i}/"
-      response = CalenderSpider.process(url)
-      i+= 1
-    end
-  end
-  # GET /events/1 or /events/1.json
+
   def berlins_show
```
Comment (MuhammadRaza29) [inline app/controllers/events_controller.rb]:
```suggestion
```

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -1,38 +1,14 @@
...
-    @events = Event.where(secondtype: "berlinevent")
+    @events = Event.where(websource: "berlinevent")
   end
 
   def calenders_show
```
Comment (MuhammadRaza29) [inline app/controllers/events_controller.rb]:
```suggestion
```

File: config/initializers/sidekiq.rb [hunk 4]
```diff
@@ -1,17 +1,6 @@
...
-  
-  Sidekiq.configure_client do |config|
-    config.redis = { url: "redis://#{redis_host}:#{redis_port}/#{redis_db}" }
-  end
+Sidekiq.configure_server do |config|
+  config.redis = { url: ENV.fetch('REDIS_URL', 'redis://localhost:6379/1') }
+end
+Sidekiq.configure_client do |config|
+  config.redis = { url: ENV.fetch('REDIS_URL', 'redis://localhost:6379/1') }
+end
```
Comment (MuhammadRaza29) [inline config/initializers/sidekiq.rb]:
```suggestion
end

```

File: config/routes.rb [hunk 5]
```diff
@@ -16,8 +12,4 @@
   match 'search_date/events' , to: 'events#search_date' ,via: :get
 
   root to: 'events#index'
-
- 
-  
-
 end
```
Comment (MuhammadRaza29) [inline config/routes.rb]:
```suggestion
end

```

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (moizansari22) [reply app/controllers/events_controller.rb]: yes added the functionality . now index controller is using params to filter events based on parameters .

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (moizansari22) [reply app/controllers/events_controller.rb]: removed the extra lines.

File: app/controllers/events_controller.rb (hunk 3 above)
Comment (moizansari22) [reply app/controllers/events_controller.rb]: removed calender_show which was extra controller.

File: config/initializers/sidekiq.rb (hunk 4 above)
Comment (moizansari22) [reply config/initializers/sidekiq.rb]: added one extra line at the end of code.

**Review by MuhammadRaza29 — APPROVED** [review]

**Pull request merged**